pkg/gateway/filter/handler/grpc/handler.go

```diff
@@ -1434,47 +1434,57 @@ func (s *server) Search(ctx context.Context, req *payload.Search_Request) (res *
-		addrs := make([]string, 0, len(targets)+len(s.DistanceFilters))
-		addrs = append(addrs, s.DistanceFilters...)
-		for _, target := range targets {
-			addrs = append(addrs, fmt.Sprintf("%s:%d", target.GetHost(), target.GetPort()))
+	filterConfigs = req.GetConfig().GetEgressFilters()
+	if filterConfigs != nil || s.DistanceFilters != nil {
+		filters := make([]*config.DistanceFilterConfig, 0, len(filterConfigs)+len(s.DistanceFilters))
+		filters = append(filters, s.DistanceFilters...)
+		for _, c := range filterConfigs {
+			filters = append(filters, &config.DistanceFilterConfig{
+				Addr:  net.JoinHostPort(c.GetTarget().GetHost(), uint16(c.GetTarget().GetPort())),
+				Query: c.Query.GetQuery(),
+			})
 		}
-		c, err := s.egress.Target(ctx, addrs...)
-		if err != nil {
-			err = status.WrapWithUnavailable(
-				fmt.Sprintf(vald.SearchRPCName+" API egress filter targets %v not found", addrs),
-				err,
-				&errdetails.RequestInfo{
-					RequestId:   req.GetConfig().GetRequestId(),
-					ServingData: errdetails.Serialize(req),
-				},
-				&errdetails.BadRequest{
-					FieldViolations: []*errdetails.BadRequestFieldViolation{
-						{
-							Field:       "vectorizer targets",
-							Description: err.Error(),
+		for _, filterConfig := range filters {
+			c, err := s.egress.Target(ctx, filterConfig.Addr)
+			if err != nil {
+				err = status.WrapWithUnavailable(
+					fmt.Sprintf(vald.SearchRPCName+" API egress filter target %v not found", filterConfig.Addr),
+					err,
+					&errdetails.RequestInfo{
+						RequestId:   req.GetConfig().GetRequestId(),
+						ServingData: errdetails.Serialize(req),
+					},
+					&errdetails.BadRequest{
+						FieldViolations: []*errdetails.BadRequestFieldViolation{
+							{
+								Field:       "vectorizer target",
+								Description: err.Error(),
+							},
 						},
 					},
-				},
-				&errdetails.ResourceInfo{
-					ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/vald.v1." + vald.SearchRPCName,
-					ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, s.name, s.ip),
-				}, info.Get())
-			log.Warn(err)
-			if span != nil {
-				span.RecordError(err)
-				span.SetAttributes(trace.StatusCodeUnavailable(err.Error())...)
-				span.SetStatus(trace.StatusError, err.Error())
+					&errdetails.ResourceInfo{
+						ResourceType: errdetails.ValdGRPCResourceTypePrefix + "/vald.v1." + vald.SearchRPCName,
+						ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, s.name, s.ip),
+					}, info.Get())
+				log.Warn(err)
+				if span != nil {
+					span.RecordError(err)
+					span.SetAttributes(trace.StatusCodeUnavailable(err.Error())...)
+					span.SetStatus(trace.StatusError, err.Error())
+				}
+				return nil, err
 			}
-			return nil, err
-		}
-		for i, dist := range res.GetResults() {
-			d, err := c.FilterDistance(ctx, dist)
+			dist := res.GetResults()
+			q := &payload.Filter_Query{
+				Query: filterConfig.Query,
+			}
+			d, err := c.FilterDistance(ctx, &payload.Filter_DistanceRequest{
```

Comment:
Here `FilterDistance` is applied but is `FilterVector` applied to the search result anywhere?